brave/brave-core
# Added web discovery dialog

fix brave/brave-browser#18488

Launched dialog for 3rd, 10th and 20th visits for brave search page
when user doesn't enable web discovery option yet or doesn't check
don't ask checkbox before.

Windows:
![image](https://user-images.githubusercontent.com/6786187/135974460-e3614172-6e13-45ae-94ba-f7e4bb961683.png)
![image](https://user-images.githubusercontent.com/6786187/135974534-09e9d1f8-acbb-498e-a5cf-b79f119e5bb9.png)

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
`npm run test brave_unit_tests -- --filter=WebDiscoveryTest.*`

1. Launch browser with clean profile and check brave search is not set as a default search provider
2. Load search.brave.com several times
3. Load brave://pref-internals and check `brave_search_visit_count`  is still zero.
4. Set brave search as a default search provider
5. Check web discovery dialog is visible for 3rd, 10th and 20th search.brave.com loading
6. Check dialog is not visible anymore after 20th loading
7. After web discovery is enabled or don't ask is checked, dialog should not be visible
8. Also check visit count is increased only when option is disabled and brave is default search provider
9. When learn more link is clicked, new foreground tab is created right after the current tab and it loads https://www.brave.com/browser/privacy/#web-discovery-project. but dialog is still opened in prev tab.
